src/cursor/cur_layered.c

```diff
@@ -2057,53 +2028,158 @@ __clayered_next_random(WT_CURSOR *cursor)
+ *     Apply a modify on a leader node.
+ */
+static int
+__clayered_modify_leader(WT_CURSOR *cursor, const WT_ITEM *key, WT_MODIFY *entries, int nentries)
+{
+    WT_CURSOR_LAYERED *clayered = (WT_CURSOR_LAYERED *)cursor;
+    WT_CURSOR *c = clayered->stable_cursor;
+
+    c->set_key(c, key);
+    WT_RET(c->search(c));
+    WT_RET(c->modify(c, entries, nentries));
+    WT_RET(__cursor_needvalue(c));
+
+    clayered->current_cursor = c;
+
+    return (0);
+}
+
+/*
+ * __clayered_modify_follower_insert --
+ *     When we're modifying on a follower, it's possible that the ingest table doesn't have an
+ *     existing value to apply the modify on top of. In this case, we must read the value from the
+ *     stable table, insert it to the ingest table, then apply the modify on that. This function
+ *     does the read/insert part of that work.
+ */
+static int
+__clayered_modify_follower_insert(
+  WT_CURSOR_LAYERED *clayered, const WT_ITEM *key, WT_MODIFY *entries, int nentries)
+{
+    WT_CURSOR *ingest = clayered->ingest_cursor;
+    WT_CURSOR *stable = clayered->stable_cursor;
+
+    if (stable == NULL)
+        /*
+         * If we have no stable table, or we can't find the key in the stable table, then there's
+         * nothing we can do -- the user is calling modify, but we can't get a base value.
+         */
+        return (WT_NOTFOUND);
+
+    /* Pull the base value out of the stable table. */
+    stable->set_key(stable, key);
+    WT_RET(stable->search(stable));
+    WT_RET(__cursor_needkey(stable));
+    WT_RET(__cursor_needvalue(stable));
+
+    /* Insert that base value in the ingest table, then apply our modifications. */
+    ingest->set_key(ingest, &stable->key);
+    ingest->set_value(ingest, &stable->value);
+    WT_RET(__wt_modify_apply_api(ingest, entries, nentries));
+    WT_RET(ingest->update(ingest));
+    WT_RET(__cursor_needvalue(ingest));
+
+    return (0);
+}
+
+/*
+ * __clayered_modify_follower --
+ *     Apply a modify on a follower node.
+ */
+static int
+__clayered_modify_follower(WT_CURSOR *cursor, const WT_ITEM *key, WT_MODIFY *entries, int nentries)
+{
+    WT_DECL_RET;
+    WT_CURSOR_LAYERED *clayered = (WT_CURSOR_LAYERED *)cursor;
+    WT_CURSOR *ingest = clayered->ingest_cursor;
+
+    /* Do we have a base value in the ingest table? */
+    ingest->set_key(ingest, key);
+    ret = ingest->search(ingest);
+    WT_RET(__cursor_localvalue(ingest));
+
+    /* Manually handle deletes. */
+    if (ret == 0 && __wt_clayered_deleted(&ingest->value))
+        ret = WT_NOTFOUND;
+
+    /*
+     * If we have a base value, just delegate the work to the constituent cursor. Otherwise, we need
+     * to get a base value somehow -- pull it out of the stable table.
+     */
+    if (ret == WT_NOTFOUND)
+        WT_RET(__clayered_modify_follower_insert(clayered, key, entries, nentries));
+    else if (ret != 0)
+        WT_RET(ret);
+    else
+        WT_RET(ingest->modify(ingest, entries, nentries));
```

Comment:
np: it seems like we don't really need else here
```suggestion
    WT_RET(ingest->modify(ingest, entries, nentries));
```

Reply:
Unfortunately, we do need the else -- `__clayered_modify_follower_insert` has to do modify work internally, so we can't fall through here (and do it a second time).

Reply:
My bad, thanks for the clarification!